extensions/panache/panache-hibernate-common/runtime/src/main/java/io/quarkus/panache/hibernate/common/runtime/PanacheJpaUtil.java

```diff
@@ -86,33 +86,31 @@ public static boolean isNamedQuery(String query) {
         return query.charAt(0) == '#';
     }
 
-    public static String createCountQuery(Class<?> entityClass, String query, int paramCount) {
-        if (query == null)
-            return "SELECT COUNT(*) FROM " + getEntityName(entityClass);
+    public static String createQueryForCount(Class<?> entityClass, String query, int paramCount) {
+        if (query == null || query.isEmpty())
+            return "FROM " + getEntityName(entityClass);
 
         String trimmedForAnalysis = trimForAnalysis(query);
         if (trimmedForAnalysis.isEmpty())
-            return "SELECT COUNT(*) FROM " + getEntityName(entityClass);
+            return "FROM " + getEntityName(entityClass);
 
         // assume these have valid select clauses and let them through
         if (trimmedForAnalysis.startsWith("select ")
```

Comment:
I suspect this used to allow custom queries such as `MyEntity.count("select count(...) from MyEntity` but now they would still pass, but we'd add `getResultCount()` which would always return `1`. Oh well… 

OTOH, it doesn't look like we document count queries at https://quarkus.io/guides/hibernate-orm-panache#simplified-queries nor in the javadoc, so I suppose we may not have complaining users…

Reply:
Me too. And in any case if it's not documented, it's not officially supported, so we're fine and we'll see who reports issues.